packages/documentsearch/src/providers/codemirrorsearchprovider.ts

```diff
@@ -30,30 +30,35 @@
   THE SOFTWARE.
 */
 
-import { ISearchProvider, ISearchMatch } from '../interfaces';
+import * as CodeMirror from 'codemirror';
+
+import { ISignal, Signal } from '@phosphor/signaling';
```

Comment:
Why the random reordering of imports?

Reply:
Well, this is confusing. I reordered them since I thought the general jlab convention was {externals, `@jupyterlab`, locals}. However, the convention seems to vary by package, and in `@jupyterlab/documentsearch` it's {locals, `@jupyterlab`, externals}. Which just seems weird to me, but whatever. 

I restored the original ordering.

Reply:
I think you are right that the signaling of which pattern to use is unclear. I thought the current practice was alphabetical? Who knows.. I'm generally in favor of keeping churn low to keep the git blame clean, but I know I'm a bit alone in that point of view on this project :)

Reply:
What can I say? Messy imports just bug me :P